eng/build-test-job.yml

```diff
@@ -42,52 +42,11 @@ jobs:
       continueOnError: true
 
     # Compute job name from template parameters
-    ${{ if and(eq(parameters.testGroup, 'innerloop'), eq(parameters.displayNameArgs, '')) }}:
-      name: 'build_test_p0_${{ parameters.osGroup }}${{ parameters.osSubgroup }}_${{ parameters.archType }}_${{ parameters.buildConfig }}'      
-      displayName: 'Build Test Pri0 ${{ parameters.osGroup }}${{ parameters.osSubgroup }} ${{ parameters.archType }} ${{ parameters.buildConfig }}'
-
-    ${{ if and(ne(parameters.testGroup, 'innerloop'), eq(parameters.displayNameArgs, '')) }}:
-      name: 'build_test_p1_${{ parameters.osGroup }}${{ parameters.osSubgroup }}_${{ parameters.archType }}_${{ parameters.buildConfig }}'      
-      displayName: 'Build Test Pri1 ${{ parameters.osGroup }}${{ parameters.osSubgroup }} ${{ parameters.archType }} ${{ parameters.buildConfig }}'
-
-    ${{ if and(eq(parameters.testGroup, 'innerloop'), ne(parameters.displayNameArgs, '')) }}:
-      name: 'build_test_p0_${{ parameters.displayNameArgs }}_${{ parameters.osGroup }}${{ parameters.osSubgroup }}_${{ parameters.archType }}_${{ parameters.buildConfig }}'      
-      displayName: 'Build Test Pri0 ${{ parameters.displayNameArgs }} ${{ parameters.osGroup }}${{ parameters.osSubgroup }} ${{ parameters.archType }} ${{ parameters.buildConfig }}'
-
-    ${{ if and(ne(parameters.testGroup, 'innerloop'), ne(parameters.displayNameArgs, '')) }}:
-      name: 'build_test_p1_${{ parameters.displayNameArgs }}_${{ parameters.osGroup }}${{ parameters.osSubgroup }}_${{ parameters.archType }}_${{ parameters.buildConfig }}'      
-      displayName: 'Build Test Pri1 ${{ parameters.displayNameArgs }} ${{ parameters.osGroup }}${{ parameters.osSubgroup }} ${{ parameters.archType }} ${{ parameters.buildConfig }}'
+    name: 'build_test_${{ parameters.priBuildNamePart }}${{ parameters.displayNameArgs }}_${{ parameters.osGroup }}${{ parameters.osSubgroup }}_${{ parameters.archType }}_${{ parameters.buildConfig }}'
+    displayName: 'Build Test ${{ parameters.priDisplayNamePart }}${{ parameters.displayNameArgs }} ${{ parameters.osGroup }}${{ parameters.osSubgroup }} ${{ parameters.archType }} ${{ parameters.buildConfig }}'
 
     crossrootfsDir: ${{ parameters.crossrootfsDir }}
 
-    variables:
-    # Map template parameters to command line arguments
-    - name: crossgenArg
-      value: ''
-    - ${{ if eq(parameters.readyToRun, true) }}:
-      - name: crossgenArg
-        value: 'crossgen'
-
-    - name: clangArg
```

Comment:
Was this intentionally deleted?

Reply:
I believe these shouldn't be needed here as crossgen doesn't run during managed test artifact build. The crossgen building section resides in the generated execution script and is based on checking runtime environment variables, not statically generated. I'll double-check where the environment variables are coming from but I hope it's not from this build step because we need that to remain independent on R2R vs. non-R2R build mode.

Reply:
Hmm, it will likely need some more restructuring after all as it looks like the CLRTestBatch/BashEnvironmentVariable set is indeed built in this phase. Sigh.

Reply:
@trylek, I apologize that I did not make my question clear. I was talking about deletion of the 'clangArg' parameter and not 'crossgenArg'.

Reply:
OK, great news, I found out this should be already taken care of here:

https://github.com/dotnet/coreclr/blob/088299e2b7f439dfd1f5056fdc291b9d82149fc1/eng/send-to-helix-step.yml#L45

I'm still waiting for confirmation from someone on the infra team as this seems almost too good to be true :-). This should basically mean that, once this change is passing lab testing, it should be ready to go.

Reply:
Just to make sure I know the problem, we are worried that R2R may not actually be running R2R?

Reply:
That's exactly the point. Sergiy inadvertently pointed me to this problem [it's a fact that the build test job can emit the RunCrossGen environment variable to the execution script] but now it seems to me that the send-to-helix step takes care of that by exposing the environment variable explicitly. Can you please confirm that my understanding is correct?

Reply:
To my knowledge build-test should have little to no knowledge about R2R. I believe it used to be used to crossgen the frameworks, however, that should be a small coverage gap.

Reply:
Other than that, it would be certainly helpful to have "reactive means" to double-check whether a particular run that was supposed to be R2R indeed launched the Crossgen compiler. I guess we can probably get to that through Helix logs similar to the one you shared the other day demonstrating the Crossgen PE architecture mismatch causing an arm32 failure in my change.

Reply:
Unfortunately as is now, it would require access to the logs of passing tests.

Without an annoyingly huge amount of changed, you can confirm it by failing at least one r2r test.

Reply:
Either way, to my knowledge the helix submission with r2r set should do what we expect @echesakovMSFT for a second set of eyes and opinions.

Reply:
Great, thanks, I pinged Egor offline right when he was leaving off for lunch so he's got more context now :-). I think we should be fine assuming the leading underscores somehow magically disappear as part of the syntax.

Reply:
@jashook @tryler I will take a look at this PR later today

Reply:
@trylek It's a general problem we have, and have had in the past, where we don't do a good job of ensuring that we are testing what we think we are testing. E.g., on Linux `COMPlus` variables are case-sensitive. We've had cases where we set the variables, but not in the correct case, so the stress/scenario mode didn't get enabled. Thanks for trying to ensure that we are actually testing the right thing!